src/sentry/deletions/defaults/group.py

```diff
@@ -47,8 +47,9 @@ def delete_instance(self, instance):
         return super(GroupDeletionTask, self).delete_instance(instance)
 
     def mark_deletion_in_progress(self, instance_list):
-        from sentry.models import GroupStatus
+        from sentry.models import Group, GroupStatus
 
-        for instance in instance_list:
-            if instance.status != GroupStatus.DELETION_IN_PROGRESS:
-                instance.update(status=GroupStatus.DELETION_IN_PROGRESS)
+        in_progress = Group.objects.filter(
```

Comment:
This local variable name is confusing, since upon assignment this queryset will include all of the rows in the instance list that are /not/ in progress due to the `exclude` filter. I'd just chain the `exclude` and `update` together and forgo the assignment.